src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -633,8 +639,7 @@ static int parse_request(uint8_t code, uint8_t len, const void *option, void *us
                 break;
         case SD_DHCP_OPTION_MAXIMUM_MESSAGE_SIZE:
                 if (len == 2)
-                        req->max_optlen = be16toh(*(be16_t*)option) -
-                                          - sizeof(DHCPPacket);
+                        req->max_optlen = unaligned_read_be16(option) + sizeof(DHCPPacket);
```

Comment:
This patch is semantically correct (the double minus is replaced by a single plus), but it does not really make sense. In my gut feeling, the maximum option length should be the maximum message size reduced by the DHCPPacket (header) size, not get it added.